### PR TITLE
FEDX-44 : Update to null safety

### PR DESCRIPTION
**Pull request by alishadalal-wk**
## Summary

Migrate to null safety. Run the null safety migration tool with Dart migrate and check/adjust its suggestions if needed.
Apply the migration, and fix any static errors. Do a dart format, run the tests, fixing any issues.

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by dustinlessard-wf — APPROVED** [review]

File: example/json_serializable_span.dart [hunk 1]
```diff
@@ -43,16 +43,16 @@ class JsonSerializableSpan {
...
-    fieldMap[spanIdField] = span?.context?.spanId;
-    fieldMap[sampledField] = span?.context?.sampled ?? false;
-    fieldMap[baggageField] = span?.context?.baggage;
-    fieldMap[parentSpanIdField] = span?.parentContext?.spanId ?? '';
-    fieldMap[operationField] = span?.operationName;
-    fieldMap[startField] = span?.startTime?.toUtc()?.toIso8601String();
-    fieldMap[durationField] = span?.duration?.inMilliseconds;
-    fieldMap[tagsField] = span?.tags;
-    fieldMap[logsField] = span?.logData;
+    fieldMap[traceIdField] = span.context.traceId;
```
Comment (dustinlessard-wf) [inline example/json_serializable_span.dart:46]: Hmmm... Can't span or context still be null?

**Review by dustinlessard-wf — CHANGES_REQUESTED** [review]
Need to clarify something first.

**Review by dustinlessard-wf — APPROVED** [review]

**Comment by robbecker-wf** [comment]
+10 @Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**